test/session_pool/session_pool.test.js

```diff
@@ -310,4 +314,29 @@ describe('SessionPool - testing session pool', () => {
         await sessionPool.teardown();
         expect(events.listenerCount(ACTOR_EVENT_NAMES_EX.PERSIST_STATE)).toEqual(0);
     });
+
+    test('should be able to create session with provided id', async () => {
+        sessionPool.maxPoolSize = 1;
+        await sessionPool.addSession({ id: 'test-session' });
+        const session = sessionPool.sessions[0];
+        expect(session.id).toBe('test-session');
+    });
+
+    test('should not be able to add session to the pool with id already in the pool', async () => {
+        try {
+            await sessionPool.addSession({ id: 'test-session' });
+            await sessionPool.addSession({ id: 'test-session' });
+        } catch (e) {
+            expect(e.message).toBe("Cannot add session with id 'test-session' as it already exists in the pool");
```

Comment:
you need `expect.assertions(1)` in this test. If no error is thrown, the test will pass.

Reply:
added